common/lib/protocol-definitions/src/clients.ts

```diff
@@ -5,53 +5,137 @@
 
 import { IUser } from "./users";
 
+/**
+ * A client's connection mode - either view-only ("read") or allowing edits ("write").
+ *
+ * @remarks Note: this should not be confused with user permissions.
+ * If a user does not have write privileges, they will still not be permitted to make edits, even if the
+ * connection mode is set to "write".
+ */
 export type ConnectionMode = "write" | "read";
 
+/**
+ * TODO
+ */
 export interface ICapabilities {
+	/**
+	 * Indicates if the client represents a potentially interactive session with a user (if 'true') or if it's a "system entity" (if 'false').
+	 *
+	 * @remarks
+	 *
+	 * The only "system entity" scenario at the moment is the
+	 * {@link https://fluidframework.com/docs/concepts/summarizer/ | summarizer client}.
+	 */
 	interactive: boolean;
 }
+
+/**
+ * TODO
+ */
 export interface IClientDetails {
+	/**
+	 * {@inheritDoc ICapabilities}
+	 */
 	capabilities: ICapabilities;
+
+	/**
+	 * TODO: What is this? Are there specific expected values? What does it mean for this to be undefined?
+	 */
 	type?: string;
+
 	/**
 	 * If the environment needs to specify multiple properties which gives info about the environment, then
-	 * it should be in particular format like: "prop1:val1;prop2:val2;prop3:val3"
+	 * it should be in particular format like: `prop1:val1;prop2:val2;prop3:val3`.
+	 *
+	 * TODO: What does it mean for this to be undefined?
 	 */
 	environment?: string;
+
+	/**
+	 * TODO: What is this? What does it mean for this to be undefined?
+	 */
 	device?: string;
 }
 
+/**
+ * TODO
+ */
 export interface IClient {
+	/**
+	 * {@inheritDoc ConnectionMode}
+	 */
 	mode: ConnectionMode;
+
+	/**
+	 * {@inheritDoc IClientDetails}
+	 */
 	details: IClientDetails;
+
+	/**
+	 * TODO: What is this? Are there specific values we expect?
+	 */
 	permission: string[];
+
+	/**
+	 * The user associated with this client.
+	 *
+	 * @remarks There may be more than 1 client associated with the same user.
+	 */
 	user: IUser;
+
+	/**
+	 * TODO: What is this? Are there specific values we expect?
+	 */
 	scopes: string[];
```

Comment:
scopes describe what user can do. For example, even with "read" connection, one of the scopes can say that user has write permissions to a file. Another scope scope could tell if client could produce summaries.

Reply:
Are the values generally expected to be flags of `ScopeType`? Is this expressed as a string because we expect clients to define their own scopes? Or just for cross-version compat?

Reply:
I think you are right, but I'd trace code to validate that. And if we can clarify type (i.e. make it ScopeType[]), even better!

Reply:
The ScopeType values appear to be the only values we look in the `fluidframework` repo here, and only places in the `bohemia` repo where this array is provided to us (for token generation) goes through an API that already only accepts `ScopeType[]`, so unless I'm missing something, I think we can safely start migrating to `ScopeType` here and in `ITokenClaims`.